# Task to do :

**Preprocessed the links_small dataset for the upcoming EDA** :( remove duplicates, handle missing value and mistype columns, remove outliers )

The movies dataset -> https://www.kaggle.com/rounakbanik/the-movies-dataset

# 1. Loading the data

>## First we import all the useful libraries 
>- Path for filepath managing
>- pandas and numpy for dataframe manip 
>- and for dataviz seaborn and matplotlib

In [1]:
from pathlib import Path

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

>##  We organise our different files directories

In [2]:
DATA_DIR = Path('../data')
RAW_DIR = DATA_DIR / 'raw'
PROCESSED_DIR = DATA_DIR / 'processed'

>##  We load the different datasets into dataframes using Pandas

In [3]:
raw_links_df = pd.read_csv(RAW_DIR / 'links_small.csv')
display(raw_links_df.sample(3))

movieId  imdbId   tmdbId
1834     2318  147612  10683.0
3227     4026  223530  25660.0
1040     1286   81534  16633.0

# 2. Preprocessing of the data 

This data contains the TMDB and IMDB IDs of a small subset of 9,000 movies of the Full Dataset.<br>
Columns contents:
- **movieId:** The id of the movie. Movie ids are consistent between `ratings_small.csv`, `movies_metadata.csv`, `credits.csv`, `keywords.csv`and `links_small.csv` (i.e., the same id refers to the same movie across these data files).
- **imdbId:** The identifier for the Internet Movie Database (IMDb) [with prefix 'tt', 'nm', 'co', 'ev', 'ch' or 'ni']. 
- **tmdbId:** Identifier for a movie at The Movie Database (TMDb). 

## 2.1. Information on the data

In [4]:
raw_links_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9125 entries, 0 to 9124
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   movieId  9125 non-null   int64  
 1   imdbId   9125 non-null   int64  
 2   tmdbId   9112 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 214.0 KB


In [5]:
raw_links_df.duplicated().value_counts()

False    9125
dtype: int64

In [6]:
raw_links_df.sample(3)

movieId   imdbId    tmdbId
3678     4691    87130   42033.0
8351   103341  1213663  107985.0
586       691   117104   40001.0

> The link table:
>- has 9125 rows and 3 columns:
    >    - only column named : tmdbId has some null value (13) **( we will just drop thoses rows since 13 over 9112 is very insignificant )** 
    >    - No duplicated 
>- 1 float and 2 integers 
>- The float columns tmdbId is an id so it should also be an integer **( we will handle this in next part )**

## 2.2. Handling missing, incorrect and invalid data

#### a. missing data 

In [7]:
raw_links_df.shape

(9125, 3)

In [8]:
raw_links_df.isnull().sum()

movieId     0
imdbId      0
tmdbId     13
dtype: int64

> 13 over 9125 seem insignificant that is why we will drop thoses null values

In [9]:
processed_links_df = raw_links_df.dropna().copy()

#### b. mistype

In [10]:
processed_links_df = processed_links_df.astype({'tmdbId':'int'})


In [11]:
processed_links_df.to_csv(PROCESSED_DIR/ 'links.csv', index=False)